crates/server/src/server/request_handler.rs

```diff
@@ -118,6 +118,20 @@ impl ResponseInfo {
         header.set_response_code(ResponseCode::ServFail);
         header.into()
     }
+
+    pub(crate) fn response_code_error(id: u16, code: ResponseCode) -> Self {
+        let mut header = Header::new();
+        header.set_id(id);
+        header.set_response_code(code);
+        header.into()
+    }
+
+    pub(crate) fn unknown(id: u16) -> Self {
+        let mut header = Header::new();
+        header.set_id(id);
+        header.set_response_code(ResponseCode::Unknown(0));
```

Comment:
There are predefined ranges for the `Unknown` enum variant: https://github.com/NordSecurity/trust-dns/blob/main/crates/proto/src/op/response_code.rs#L125-L130